scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala

```diff
@@ -409,7 +409,7 @@ trait ClassLikeSupport:
       if  argument.symbol.flags.is(Flags.Covariant) then "+"
       else if argument.symbol.flags.is(Flags.Contravariant) then "-"
       else ""
-    val name = argument.symbol.normalizedName
+    val name = argument.symbol.normalizedName.takeWhile(_ != '$')
```

Comment:
One can use `$` in type name, eg. `class A[A$B]` compiles fine.

Reply:
Aleksander told me that using `$` sign in names in Scala is widely unrecommended and no one guarantees that everything works fine, though I can chagne it to be the reqex matching `_\$\d*`

Reply:
Changed to regex